cpp/src/arrow/flight/transport/grpc/grpc_server.cc

```diff
@@ -598,7 +598,7 @@ class GrpcServerTransport : public internal::ServerTransport {
     } else if (scheme == kSchemeGrpcUnix) {
       std::stringstream address;
       address << "unix:" << uri.path();
-      builder.AddListeningPort(address.str(), ::grpc::InsecureServerCredentials());
```

Comment:
Looks like this needs to be done above as well for the TLS branch

Reply:
Plus, for domain sockets, port will still be 0 right?

Reply:
> Looks like this needs to be done above as well for the TLS branch

Hm, I don't really get what you're referring to.





> Plus, for domain sockets, port will still be 0 right?

Yes, that's probably correct (however, I don't have any practical experience with domain sockets in this context).
That would mean that adding the port here does not make sense in the first place and we would need to check below that `scheme != kSchemeGrpcUnix` for issuing that _port bind specific_ error message.

What do you think?

Reply:
This is an if/else. I think this change should only apply to the TCP socket-based schemes and not the domain socket, so the opposite of the current change.

Reply:
The more specific message is now only used if the URI scheme is a host/port combination. I hope this logic corresponds to what you had in mind.